pkg/registries/registries.go

```diff
@@ -4,20 +4,34 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/containers/image/pkg/sysregistriesv2"
+	"github.com/containers/image/v5/pkg/sysregistriesv2"
 	apioperatorsv1alpha1 "github.com/openshift/api/operator/v1alpha1"
 )
 
-// scopeMatchesRegistry returns true if a scope value (as in sysregistriesv2.Registry.Prefix / sysregistriesv2.Endpoint.Location)
-// matches a host[:port] value in reg.
-func scopeMatchesRegistry(scope, reg string) bool {
-	if reg == scope {
+// regMatchesScope returns true if a subScope value (as in sysregistriesv2.Registry.Prefix / sysregistriesv2.Endpoint.Location)
+// matches a host[:port] value in superScope or if it is a sub-scope of superScope.
+func scopeIsNestedInsideScope(subScope, superScope string) bool {
+	match := false
+	if superScope == subScope {
 		return true
 	}
-	if len(scope) > len(reg) {
-		return strings.HasPrefix(scope, reg) && scope[len(reg)] == '/'
+	// return true if a scope value matches a host[:port] value in reg
```

Comment:
```suggestion
	// return true if subScope defines a namespace/repo inside (non-wildcard) superScope
```

because this applies to `host.com/foo` vs. `host.com/foo/bar` as well.

---

This probably should be an `if … > … && HasPrefix && == '/' { return true }`, because the code below unconditionally overwrites `match = strings.HasSuffix`. AFAICS this works correctly but with 4 possible code paths through it is not _obvious_.

Alternatively split the code into distinct “`superScope` is/isn’t a wildcard” paths; so that this check does not happen for `*.foo/bar` vs `*.foo/bar/baz`. OTOH at least one of the inputs would have to be invalid, so it doesn’t _matter_ at all if all inputs go through the validation function.

Reply:
Fixed